Altis_Life.Altis/dialog/function/fn_useGangBank.sqf

```diff
@@ -6,34 +6,34 @@
     Description:
     Withdraws money from the gang bank.
 */
-private ["_value"];
-_value = parseNumber(ctrlText 2702);
-_gFund = GANG_FUNDS;
-group player setVariable ["gbank_in_use_by",player,true];
+params [
+    ["_deposit",false,[false]]
+];
+
+private _value = parseNumber(ctrlText 2702);
+private _gFund = GANG_FUNDS;
+if ((time - life_action_delay) < 0.5) exitWith {hint localize "STR_NOTF_ActionDelay"};
 
 //Series of stupid checks
 if (isNil {(group player) getVariable "gang_name"}) exitWith {hint localize "STR_ATM_NotInGang"}; // Checks if player isn't in a gang
 if (_value > 999999) exitWith {hint localize "STR_ATM_WithdrawMax";};
-if (_value < 0) exitWith {};
+if (_value < 1) exitWith {};
 if (!([str(_value)] call TON_fnc_isnumber)) exitWith {hint localize "STR_ATM_notnumeric"};
-if (_value > _gFund) exitWith {hint localize "STR_ATM_NotEnoughFundsG"};
+if ((_deposit && _value > CASH) || (!_deposit && _value > _gFund)) exitWith {hint localize "STR_ATM_NotEnoughCash"};
 if (_val < 100 && _gFund > 20000000) exitWith {hint localize "STR_ATM_WithdrawMin"}; //Temp fix for something.
-if ((group player getVariable ["gbank_in_use_by",player]) != player) exitWith {hint localize "STR_ATM_WithdrawInUseG"}; //Check if it's in use.
 
-_gFund = _gFund - _value;
-CASH = CASH + _value;
-group player setVariable ["gang_bank",_gFund,true];
+life_action_delay = time;
```

Comment:
Would be better to have this as the very last statement; this function is running unscheduled